mne/io/reference.py

```diff
@@ -386,20 +390,36 @@ def set_eeg_reference(inst, ref_channels='average', copy=True,
             return inst, None
 
     inst = inst.copy() if copy else inst
+    # if ch_type is 'auto', search through list to find first reasonable
+    # reference-able channel type.
+    possible_types = ['eeg', 'ecog', 'seeg']
+    if ch_type == 'auto':
+        for type in possible_types:
+            if type in inst:
+                ch_type = type
+                break
+        logger.info('%s channel type selected for re-referencing' % ch_type)
+    # if auto comes up empty, or the user specifies a bad ch_type.
+    if ch_type not in possible_types:
+        raise ValueError('None of the possible channel types for '
+                         're-referencing were found.')
+
+    ch_dict = {ch_type: True, 'meg': False, 'ref_meg': False}
+    eeg_idx = pick_types(inst.info, **ch_dict)
+    ch_sel = [inst.ch_names[i] for i in eeg_idx]
 
     if ref_channels == 'average' and not projection:  # apply average reference
         logger.info('Applying average reference.')
-        eeg_idx = pick_types(inst.info, eeg=True, meg=False, ref_meg=False)
-        ref_channels = [inst.ch_names[i] for i in eeg_idx]
+        ref_channels = ch_sel
 
     if ref_channels == []:
         logger.info('EEG data marked as already having the desired reference. '
                     'Preventing automatic future re-referencing to an average '
                     'reference.')
     else:
-        logger.info('Applying a custom EEG reference.')
+        logger.info('Applying a custom reference.')
```

Comment:
I would at the top `from ..defaults import DEFAULTS` then here use `DEFAULTS['titles'][type_]` with `%s` in the str